python-lib/robust_session.py

```diff
@@ -24,12 +26,14 @@ def __init__(self, session=None, status_codes_to_retry=None, max_retries=1, base
         self.connection_kwargs = {}
         self.connection_library = None
         self.attempt_session_reset_on_403 = attempt_session_reset_on_403
+        self.default_headers = {}
 
-    def update_settings(self, session=None, status_codes_to_retry=None, max_retries=None, base_retry_timer_sec=None):
+    def update_settings(self, session=None, status_codes_to_retry=None, max_retries=None, base_retry_timer_sec=None, default_headers=None):
```

Comment:
🧑‍🍳 suggestion: 
```suggestion
    def update_settings(self, session=None, status_codes_to_retry=None, max_retries=None, base_retry_timer_sec=None, default_headers=self.default_headers):
```
and remove l.36

Reply:
My linter tells me I can't use `self` there...

Reply:
Fine then, no problem !